### PR TITLE
Add -vet option to umka cli thing

### PR DESCRIPTION
**Pull request by marekmaskarinec**
You can now only check, if program has no errors without running it.

### CODE REVIEW

**Review by vtereshkov — CHANGES_REQUESTED** [review]
A good idea! But the new parameter has to be allowed at any place, not necessarily at `argv[2]`. Please also change `compileOnly` to `bool` and follow the curly bracket style used in  my C files.

**Comment by Mazdaywik** [comment]
Commit b5b92b3 removes Vim syntax file.

**Comment by marekmaskarinec** [comment]
Oh, sorry. Forgot about the brackets. I will fix it tomorrow.

**Comment by marekmaskarinec** [comment]
@Mazdaywik yes, that is wanted. The syntax file was never merged in the first place and I made a separate [repository](https://github.com/marekmaskarinec/vim-umka) for it.

**Comment by marekmaskarinec** [comment]
@vtereshkov it should all be fixed now

**Review by Mazdaywik — CHANGES_REQUESTED** [review]
Please change tabulations to spaces.

File: src/umka.c [hunk 1]
```diff
@@ -98,7 +98,7 @@ int main(int argc, char **argv)
     else
     {
         if (compileOnly)
-				{
+        {
```
Comment (Mazdaywik) [inline src/umka.c:101]: Ok.

**Review by Mazdaywik — APPROVED** [review]
@vtereshkov should approve it.

**Review by vtereshkov — APPROVED** [review]
@marekmaskarinec I approve your pull request, though I still have doubts about reading the command-line arguments correctly. So I'll merge and test it. I also think of renaming `vet` into `check`, as this command just runs the compiler and does not use extra heuristics involved by Go's `vet`.

**Comment by marekmaskarinec** [comment]
Oh. I'm sorry about the libumka.wasm file. I've been experimenting a bit and staged it when fixing the tabs.

**Comment by vtereshkov** [comment]
@marekmaskarinec I have fixed two problems:
* My former logic had a fixed step of 2 when parsing the command-line arguments, so you couldn't parse `-vet` without specifying some additional data, like in `-stack` or `-storage`. Now the step is variable. This also allows me to construct the assembly listing name automatically.
* In the compile-only mode, you terminated the program skipping `umkaFree()`. This is a possible memory leak, though an OS would probably deallocate the memory for you.

Please check the `master` branch and report any remaining inconsistencies.

**Pull request merged**